.github/workflows/library_interop_tests.yml

```
@@ -0,0 +1,183 @@
+# This workflow performs interoperability tests across the supported runtimes of the MPL.
+name: Library Interoperability Tests
+
+on:
+  workflow_call:
+    inputs:
+      dafny:
+        description: "The Dafny version to run"
+        required: true
+        type: string
+
+jobs:
+  generateEncryptVectors:
+    strategy:
+      matrix:
+        library: [TestVectorsAwsCryptographicMaterialProviders]
+        os: [
+            # TODO just test on mac and ubuntu for now
+            # windows-latest,
+            ubuntu-latest,
+            macos-latest,
+          ]
+        languages: [java, net]
+        dotnet-version: ["6.0.x"]
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - uses: actions/checkout@v3
+      # The specification submodule is private so we don't have access, but we don't need
+      # it to verify the Dafny code. Instead we manually pull the submodule we DO need.
+      - run: git submodule update --init libraries
+
+      # Set up runtimes
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
```

```suggestion
      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
        uses: actions/setup-dotnet@v3
        if: ${{ matrix.language }} == 'net'
        with:
          dotnet-version: ${{ matrix.dotnet-version }}
```

Can we do this? I think we only need to do steps for a language if it's the language in the matrix. Ditto on the other Net-only steps, and Java-only steps.